File: algoexpert/easy/1-two_sum.go
/*
package main
// O(n^2) time | O(1) space
func TwoNumberSum(array []int, target int) []int {
	// Write your code here.

	for i :=0; i < len(array)-1; i++ {
		for j :=i+1; j < len(array); j++ {			
			if array[i] + array[j] == target {
				return []int{ array[i], array[j]}
			}
		}
	}

	return []int{}
}

// Case2: 


// Case3: 
package main
func TwoNumberSum(array []int, target int) []int {
	// Write your code here.
	for i :=0; i < len(array)-1; i++ {
		for j :=i+1; j < len(array); j++ {			
			if array[i] + array[j] == target {
				return []int{ array[i], array[j]}
			}
		}
	}

	return []int{}
}

/*

File: algoexpert/array/1-two_sum.py
## case1
def twoNumberSum(array, targetSum):
    # Write your code here.
	# Time: O(n), Space: O(n)
    nums = {}
	for num in array:
		potential_match = targetSum - num
		if potential_match in nums:
			return [potential_match, num]
		else:
			nums[num] = True
	return []

## case2
def twoNumberSum(array, targetSum):
    # Write your code here.
	# Time: O(n), Space: O(n)
    nums = {}
	for num in array:
		potential_match = targetSum - num
		if potential_match in nums:
			return [potential_match, num]
		else:
			nums[num] = True
	return []

## case3
def twoNumberSum(array, targetSum):
    # Write your code here.
	# Time: O(n), Space: O(n)
    nums = {}
	for num in array:
		potential_match = targetSum - num
		if potential_match in nums:
			return [potential_match, num]
		else:
			nums[num] = True
	return []
